### PR TITLE
[DUOS-1669][risk=no] Added data submitter console

### PR DESCRIPTION
**Pull request by lu-c**
Updated to add data submitter header link, console page.

Currently, it seems like the data submitter role does not yet exist (at least from the consent swagger-ui that role id was not indicated under the users tab), but this pr could be changed to make it fit when if there are discrepancies!

Overall, there were a few more changes to this PR to try to make updating roles simpler, by moving consolidating them a little bit more.

- Moved the tab configs out of the component level so it is only rendered once (since it is a static variable atm)
- Added field "isRendered" to the tab config to remove the manual declarations of roles (ie: `isChairPerson` and `isResearcher`... etc)
- Added new header tab config for data submitter
- Added data submitter role to the utils
- Modified redirection navigation route to default to use the tabConfigs to determine which page to load when the user first logs in (ie: first console, first link). States that did not exist as part of the console were kept at the ends.

### CODE REVIEW

**Comment by connorlbark** [comment]
I like the changes here! Makes it a lot cleaner - do we want to also add the ability to make someone a data submitter from the admin console? If so, I could see that being here or in another ticket - what do you think?

**Comment by lu-c** [comment]
> I like the changes here! Makes it a lot cleaner - do we want to also add the ability to make someone a data submitter from the admin console? If so, I could see that being here or in another ticket - what do you think?

What is the admin console? Is that in teh UI? Or do you mean through the swagger ui? If the former, I'd like to clear it with the team first, just to see if there's some security concern, but it is a good idea! (always prefer doing stuff through a pretty ui 😊 )

**Comment by connorlbark** [comment]
> What is the admin console? Is that in teh UI? Or do you mean through the swagger ui? If the former, I'd like to clear it with the team first, just to see if there's some security concern, but it is a good idea! (always prefer doing stuff through a pretty ui 😊 )

Ohhh sorry I meant the admin edit user page! it would just make it easier to assign people to being data submitters. maybe let's ask tmrw at standup !

File: src/components/DuosHeader.js [hunk 1]
```diff
@@ -67,6 +67,72 @@ const styles = {
...
+  },
+  {
+    label: 'DS Console',
+    link: '/data_submission_form',
+    search: 'data_submission_form',
+    children: [
+      { label: 'Datasets', link: '/data_submission_form' }
+    ],
+    isRendered: (user) => user.isDataSubmitter
+  }
```
Comment (shaemarks) [inline src/components/DuosHeader.js:133]: I think we should move the researcher console tab config to the end of this array if we want Data Submitters to see the Data Submitter console first. Every user has a researcher role, so the `find` function that gets the first console would hit the researcher console first.

**Review by shaemarks — COMMENTED** [review]
Changes look great, and code is a lot cleaner! However, I wasn't able to give myself the Data Submitter role to test locally (the role does exist on the backend as of [DUOS-1974](https://broadworkbench.atlassian.net/browse/DUOS-1974), but it looks like a little more backend work is necessary for users to receive the role)

File: src/components/DuosHeader.js (hunk 1 above)
Comment (lu-c) [reply src/components/DuosHeader.js:133]: gotcha, I agree! Moving the researcher block to the end of the array 👍

**Review by shaemarks — APPROVED** [review]
Changes look good to me! I was able to give myself the DataSubmitter role via the `PUT /api/user/{userId}` endpoint and access the console

**Review by connorlbark — APPROVED** [review]
👍 works locally on my user who is a data submitter!

**Pull request merged**